web/crux-ui/src/pages/products/[productId]/versions/[versionId]/deployments/[deploymentId]/instance/[instanceId].tsx

```diff
@@ -0,0 +1,224 @@
+        onClick={() => setViewState('json')}
+        className="mx-8"
+        heightClassName="pb-2"
+      >
+        {t('json')}
+      </DyoButton>
+    </div>
+  )
+
+  return (
+    <Layout title={t('common:image')}>
+      <PageHeading pageLink={pageLink} sublinks={sublinks} />
+
+      <DyoCard className="p-4">
+        <div className="flex mb-4 justify-between items-start">
+          <DyoHeading element="h4" className="text-xl text-bright">
+            {instance.image.name}
+            {instance.image.name !== state.config?.name ? ` (${state.config?.name})` : null}
+          </DyoHeading>
+
+          {getViewStateButtons()}
+        </div>
+        {viewState === 'editor' && <ImageConfigFilters onChange={setFilters} />}
+      </DyoCard>
+
+      {viewState === 'editor' && (
+        <DyoCard className="flex flex-col mt-4 px-4 w-full">
+          <CommonConfigSection
+            disabled={!deploymentState.mutable}
+            selectedFilters={filters}
+            config={state.config}
+            onChange={onChange}
+            editorOptions={editorState}
+            fieldErrors={fieldErrors}
+            secrets="value"
+            definedSecrets={state.definedSecrets}
+            publicKey={deployment.publicKey}
+          />
+          <CraneConfigSection
+            disabled={!deploymentState.mutable}
+            selectedFilters={filters}
+            config={state.config}
+            onChange={onChange}
+            editorOptions={editorState}
+          />
+          <DagentConfigSection
+            disabled={!deploymentState.mutable}
+            selectedFilters={filters}
+            config={state.config}
+            onChange={onChange}
+            editorOptions={editorState}
+          />
+        </DyoCard>
+      )}
+
+      {viewState === 'json' && (
+        <DyoCard className="flex flex-col mt-4 p-4 w-full">
+          {jsonError ? (
+            <DyoMessage message={jsonError} className="text-xs italic w-full mb-2" messageType="error" />
+          ) : null}
+          <EditImageJson
+            config={state.config}
+            editorOptions={editorState}
+            onPatch={onChange}
+            onParseError={err => setJsonError(err?.message)}
+            convertConfigToJson={imageConfigToJsonContainerConfig}
+          />
+        </DyoCard>
+      )}
+    </Layout>
+  )
+}
+
+export default InstanceDetailsPage
+
+const getPageServerSideProps = async (context: NextPageContext) => {
+  const { productId, versionId, deploymentId, instanceId } = context.query
+
+  const crux = cruxFromContext(context)
+  const product = await crux.products.getById(productId as string)
+  const version = await crux.versions.getById(versionId as string)
+  const deploymentDetails = await crux.deployments.getById(deploymentId as string)
+  const node = await crux.nodes.getNodeDetails(deploymentDetails.nodeId)
+
+  const deployment: DeploymentRoot = {
+    ...deploymentDetails,
+    product,
+    version,
+    node,
```

Comment:
Why do we need the node here?

Reply:
For `DeploymentRoot` because it's used by `useDeploymentState`.

Reply:
True. Sorry.